client/spring-zeebe/src/main/java/io/camunda/zeebe/spring/client/annotation/value/factory/ReadAnnotationValueConfiguration.java

```diff
@@ -19,8 +20,8 @@ public ReadZeebeDeploymentValue readZeebeDeploymentValue(final ZeebeExpressionRe
   }
 
   @Bean
-  public ReadZeebeWorkerValue readZeebeWorkerValue(final ZeebeExpressionResolver resolver) {
-    return new ReadZeebeWorkerValue(resolver);
+  public ReadZeebeWorkerValue readZeebeWorkerValue(final ZeebeExpressionResolver resolver, final ZeebeClientProperties zeebeClientProperties) {
```

Comment:
I don't like that we need the properties now available here - does not smell super good. However, I also don't have another great idea to get rid of it 🤔

Reply:
We can create a custom interface and extract the worker's configuration method to this new interface. And then implement this interface in the `ZeebeClientConfigurationProperties` class 🤔  I will try it and propose it in the PR :)

Reply:
I've played a little with the code and maybe shall we introduce some sort of `ZeebeWorker` annotation customizer interface? So, it would work something like this:
1. We implement this interface in the `ZeebeClientConfigurationProperties` with default order.
2. We autowire a list of these interface's implementations into the `ZeebeWorkerAnnotationProcessor` class
3. We go through this list
Thus, end users get the ability to provide their own customizations if needed and in our code, there will be no dependency around properties at all. Also, end users could override the order (via `@Order` annotation) of these customizations and could control this process thoroughly.

Reply:
I definitely like to have the switch if a worker is enabled or not in the `ZeebeWorkerAnnotationProcessor` - this feels like the right place. Having an Customizer interface sounds OK - maybe you could push the code you (@aivinog1) have in mind than I am happy to have a look?

Reply:
Hey @berndruecke! I've changed the approach: right now I'm using customizers instead of direct dependency on the `ZeebeClientConfiguration`. Can you take a look? :) From my perspective, it is cleaner than the previous one :)